### PR TITLE
Legion go/rog ally audio tuning

### PR DESCRIPTION
**Pull request by matte-schwartz**
The commit changes are the framework for the $HOME/.config/easyeffects directory and sub-directory creation for both the ROG Ally and the Legion Go. 

In order to properly utilize these files, easyeffects and all its optional plugin dependencies must be installed. Running easyeffects as a background service also requires xorg-x11-server-Xvfb since it technically runs in headless server mode.

`/usr/local/bin/easyeffects-xvfb`:  this is the edited bin file for easyeffects-xvfb that gets rid of some background errors that can pop up. it also needs `sudo chmod +x` to ensure it gets run as an executable.

```
#!/bin/bash

if [[ "$1" = "start" ]]; then
    pkill Xvfb
    sleep 1
    Xvfb :43 -screen 0 1024x768x16 &
    Xvfb_PID=$!
    sleep 3
    if kill -0 $Xvfb_PID 2>/dev/null; then
        export DISPLAY=:43
        easyeffects --gapplication-service
    else
        echo "Xvfb failed to start"
        exit 1
    fi
fi
if [[ "$1" = "stop" ]]; then
    easyeffects --quit
    pkill Xvfb
fi
```

`/usr/share/applications/com.github.wwmm.easyeffects.desktop`: the desktop file for easyeffects needs a minor edit to stop and start the background service when using the desktop Easyeffects GUI:

```
Exec=bash -c 'systemctl --user disable --now easyeffects-xvfb; easyeffects; systemctl --user enable --now easyeffects-xvfb'

```

`/etc/systemd/user/easyeffects-xvfb.service`: this is the edited background service file that gets it all running properly. there is sometimes an issue where the service would become inactive after switching from Desktop to Game Mode after editing the Easyeffects GUI. The .desktop change and the Restart function of the service file counteract that.

```
[Unit]
Description=EasyEffects inside Xvfb

[Service]
Environment="DISPLAY=:43"
Type=simple
ExecStart=/usr/local/bin/easyeffects-xvfb start
ExecStop=/usr/local/bin/easyeffects-xvfb stop
Restart=always
RestartSec=5

[Install]
WantedBy=default.target

```

the service file gets enabled with `systemctl --user enable --now easyeffects-xvfb`

### CODE REVIEW

**Comment by pastaq** [comment]
There's a pretty significant problem with using $HOME, device quirks runs as root. Is there an equivalent directory in /etc that can be used?

**Comment by matte-schwartz** [comment]
Hmm, that certainly is an issue. Now that I've got the config files sorted out, let me dig into the structure tomorrow morning and I'll update the request accordingly when I get it all working together nicely.

**Comment by matte-schwartz** [comment]
$HOME issue has been addressed. tested by cloning my updated repo with Chimera 45 unstable dd49f76 on my Legion Go. added the packages required with `sudo pikaur -S easyeffects xorg-server-xvfb lsp-plugins-lv2 calf zam-plugins-lv2` and then added easyeffects-xvfb bin and .service files to their proper locations at `/usr/local/bin/easyeffects-xvfb` and `/etc/systemd/user/easyeffects-xvfb.service` and enabled with `systemctl --user enable --now easyeffects-xvfb`

**Comment by fennectech** [comment]
Will this make the ally louder?   I’ve noticed it gets quite a bit louder in windows.

**Comment by matte-schwartz** [comment]
> Will this make the ally louder? I’ve noticed it gets quite a bit louder in windows.

Yes it should. Ally audio tuning should match Windows thanks to Ripplingsnake.

**Comment by pastaq** [comment]
Hmm, I don't think this solution will work for all scenarios. Initial installation may not have a gamer user yet when this is run, or if it will be associated with UID 1000 running in an archiso environment. Looking at the easy-effects readme it does specify using the home directory so we'll need to think about this a bit.

We can definitely store the presets here. I think we'll need a user service that can install the files if not present. I'd be a lot more comfortable with that anyway as I prefer not to have root messing around in home.

**Comment by pastaq** [comment]
Whats the status of this PR with the changes we discussed?

**Comment by matte-schwartz** [comment]
> Whats the status of this PR with the changes we discussed?

I put this on the backburner so I could do more in-game testing with EasyEffects and Xvfb, and I think this specific method comes with too many tradeoffs to be considered for implementation.

A few examples: changing the TDP significantly can cause the audio to temporarily glitch out, there’s occasionally a crackle that gets introduced when Suspending/Resuming that won’t go away unless you restart the Xvfb server or your device, and low TDP settings can introduce an unacceptable amount of latency.

It seemed promising at first but these outstanding issues make it harder for me to see this as being a better stock alternative to something simple like OGUI’s volume boost or an integrated Pipewire equalizer. There’s also a [Pipewire convolver plugin](https://docs.pipewire.org/page_module_filter_chain.html) that may function similarly to EasyEffects without the added steps of a headless server like Xvfb. I just learned about this Pipewire convolver plug-in recently and plan on testing it this weekend.

**Comment by fennectech** [comment]
I mean. The audio crackling exists on vanilla ally at lower TDPs.   I think most people would be understanding of this.  And having TDP controls is better than not having them.  I. Think this should be merged as is while adding a warning to users.  And if not.  A fork is a worthwhile alternative. Sent from my iPadOn 18. Nov 2023, at 22:43, matte-schwartz ***@***.***> wrote:﻿

Whats the status of this PR with the changes we discussed?

I put this on the backburner so I could do more in-game testing with EasyEffects and Xvfb, and I think this specific method comes with too many tradeoffs to be considered for implementation.
A few examples: changing the TDP significantly can cause the audio to temporarily glitch out, there’s occasionally a crackle that gets introduced when Suspending/Resuming that won’t go away unless you restart the Xvfb server or your device, and low TDP settings can introduce an unacceptable amount of latency.
It seemed promising at first but these outstanding issues make it harder for me to see this as being a better stock alternative to something simple like OGUI’s volume boost or an integrated Pipewire equalizer. There’s also a Pipewire convolver plugin that may function similarly to EasyEffects without the added steps of a headless server like Xvfb. I just learned about this Pipewire convolver plug-in recently and plan on testing it this weekend.

**Comment by matte-schwartz** [comment]
please review the new ally.sh and legion-go.sh. in order to activate the surround sink, all users need to do is set their default speaker device to 100% and then switch the audio sink within Steam Game Mode from the default audio controller (like Family)  to the new surround sink. Volume buttons should work for the new virtual surround sink.

File: usr/share/device-quirks/scripts/asus/ally/ally.sh [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+if [ "$(whoami)" != 'root' ]; then
   echo "You must be root to run this script."
   exit 1
 fi
 
-# Create an equalizer for pipewire.
-mkdir -p ~/.config/pipewire/pipewire.conf.d
-cp $DQ_PATH/scripts/asus/ally/sink-eq6.conf ~/.config/pipewire/pipewire.conf.d/sink-eq6.conf
\ No newline at end of file
+# Define the path of the legion-go.sh script and pipewire folder
```
Comment (pastaq) [inline usr/share/device-quirks/scripts/asus/ally/ally.sh:9]: Minor edit, the comment here should be for the ally.

**Comment by pastaq** [comment]
The ChimeraOS wiki will need to be updated after v45 goes live with instructions on how to use this. We'll be overhauling the wiki after v45 anyway so we can add this info then, but a draft of the steps and any needed screenshots would be most helpful. If you can post it in this thread that will ensure testing can be thorough.

**Pull request merged**